spark3/src/main/java/org/apache/iceberg/spark/actions/BaseSnapshotTableSparkAction.java

```diff
@@ -168,6 +168,7 @@ public SnapshotTable tableProperty(String property, String value) {
     properties.remove(LOCATION);
     properties.remove(TableProperties.WRITE_METADATA_LOCATION);
     properties.remove(TableProperties.WRITE_NEW_DATA_LOCATION);
+    properties.remove(TableProperties.OBJECT_STORE_PATH);
```

Comment:
Non-blocking: it might make sense to add a comment here that we’re explicitly choosing not to bring along `OBJECT_STORE_PATH` in the snapshot?

Either a comment, or possibly updating the ObjectStorageLocationProvider docs / snapshot docs with this detail would be great 🙂. Documentation updates can be done in a separate PR of course (and happy to assist there if you’d like).